aten/src/ATen/native/Convolution.cpp

```diff
@@ -648,7 +647,7 @@ at::Tensor _convolution(
   } else if (input.device().type() == c10::DeviceType::CPU || input.device().type() == c10::DeviceType::CUDA) {
     if (params.use_cpu_depthwise3x3_winograd(input, weight)) {
       output = convolution_depthwise3x3_winograd_stub(
```

Comment:
Did this not work before?

Reply:
This is absolutely nuts.  Here's how I understand it.

This is partly a case of double inversion where I was incorrectly passing the arguments as (padding, stride) at call site here while the function was expecting (stride, padding), and again incorrectly passing them swapped a second time as (stride, padding) to calculate_conv_output_size which expected its arguments as (padding, stride) so these canceled out!  So that's that.

Now stride was also not used in _convolution_depthwise3x3_winograd beyond this point since strided convolutions are not supported by the code path so didn't matter what incorrect value it had after this point, but padding was actually used after this point, and it contained whatever value stride had, namely 1, since we gate this call on the condition that stride is 1, but padding was also set to 1 for 3x3 kernels hence why things were working!!

I shudder at the thought of how many bugs of this nature are hidden in any software of non-trivial size. :)